typescript/testsuite/testsuite_impl/basic_datastore_and_api_test/basic_datastore_and_api_test_.ts

```diff
@@ -0,0 +1,238 @@
+		this.apiImage = apiImage;
+    }
+	
+	public configure(builder: TestConfigurationBuilder): void {
+		builder.withSetupTimeoutSeconds(60).withRunTimeoutSeconds(60); //TODO (Ali) - allowed since typescript gives direct reference
+    }
+
+	public async setup(networkCtx: NetworkContext): Promise<Result<Network, Error>> { //TODO (Ali) - async?
+		
+		const [datastoreContainerCreationConfig, datastoreRunConfigFunc]: [ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>] = await getDatastoreServiceConfigurations(); //TODO (Ali) - maybe Result here
+
+		const addDatastoreServiceResult: Result<[ServiceContext, Map<string, PortBinding>], Error> = await networkCtx.addService(DATASTORE_SERVICE_ID, datastoreContainerCreationConfig, datastoreRunConfigFunc);
+		if (!addDatastoreServiceResult.isOk()) {
+			return err(addDatastoreServiceResult.error)
+		}
+		const [datastoreServiceContext, datastoreSvcHostPortBindings]: [ServiceContext, Map<string, PortBinding>] = addDatastoreServiceResult.value;
+
+		const datastoreClient: DatastoreClient = new DatastoreClient(datastoreServiceContext.getIPAddress(), DATASTORE_PORT);
+
+		const datastoreWaitForHealthyResult: Result<null, Error> = await datastoreClient.waitForHealthy(WAIT_FOR_STARTUP_MAX_POLLS, WAIT_FOR_STARTUP_DELAY_MILLISECONDS);
+		if (!datastoreWaitForHealthyResult.isOk()) {
+			return err(datastoreWaitForHealthyResult.error);
+		}
+
+		log.info("Added datastore service with host port bindings: %+v", datastoreSvcHostPortBindings);
+
+		const [apiServiceContainerCreationConfig, apiServiceRunConfigFunc]: [ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>] = await getApiServiceConfigurations(datastoreClient) //TODO (Ali) - add maybe Result here
+
+		const addAPIServiceResult: Result<[ServiceContext, Map<string, PortBinding>], Error> = await networkCtx.addService(API_SERVICE_ID, apiServiceContainerCreationConfig, apiServiceRunConfigFunc);
+		if (!addAPIServiceResult.isOk()) {
+			return err(addAPIServiceResult.error)
+		}
+		const [apiServiceContext, apiSvcHostPortBindings]: [ServiceContext, Map<string, PortBinding>] = addAPIServiceResult.value;
+
+		const apiClient: APIClient = new APIClient(apiServiceContext.getIPAddress(), API_SERVICE_PORT);
+
+		const apiWaitForHealthyResult: Result<null, Error> = await apiClient.waitForHealthy(WAIT_FOR_STARTUP_MAX_POLLS, WAIT_FOR_STARTUP_DELAY_MILLISECONDS);
+		if (!apiWaitForHealthyResult.isOk()) {
+			return err(apiWaitForHealthyResult.error);
+		}
+
+		log.info("Added API service with host port bindings: " + apiSvcHostPortBindings);
+		return ok(networkCtx);
+	}
+
+	public async run(network: Network): Promise<Result<null, Error>> { //TODO(Ali) - async?
+		// Go doesn't have generics so we have to do this cast first
```

Comment:
This isn't Go though, so make sure this comment - and all others like it - are something like `// TODO Delete when Test is genericized`